frontend/src/scenes/groups/groupsListLogic.ts

```diff
@@ -0,0 +1,64 @@
+import { kea } from 'kea'
+import api from 'lib/api'
+import { teamLogic } from 'scenes/teamLogic'
+import { urls } from 'scenes/urls'
+import { groupsModel } from '~/models/groupsModel'
+import { Group } from '~/types'
+
+import { groupsListLogicType } from './groupsListLogicType'
+interface GroupsPaginatedResponse {
+    next_url: string | null
+    previous_url: string | null
+    results: Group[]
+}
+
+export const groupsListLogic = kea<groupsListLogicType<GroupsPaginatedResponse>>({
+    path: ['groups', 'groupsListLogic'],
+    connect: { values: [teamLogic, ['currentTeamId'], groupsModel, ['groupsEnabled', 'groupTypes']] },
+    actions: () => ({
+        loadGroups: (url?: string | null) => ({ url }),
+        setTab: (tab: string) => ({ tab }),
```

Comment:
Why is tab a string here and a number below? This causes you headaches with typecasting.